packages/vue-inertia/src/index.ts

```diff
@@ -1,11 +1,42 @@
-import { Config, NamedInputEvent, RequestMethod, SimpleValidationErrors, toSimpleValidationErrors, ValidationConfig, ValidationErrors, resolveUrl, resolveMethod } from 'laravel-precognition'
+import {
+    Config,
+    NamedInputEvent,
+    RequestMethod,
+    SimpleValidationErrors,
+    toSimpleValidationErrors,
+    ValidationConfig,
+    ValidationErrors,
+    resolveUrl,
+    resolveMethod,
+    Validator,
+} from 'laravel-precognition'
 import { useForm as usePrecognitiveForm, client } from 'laravel-precognition-vue'
-import { useForm as useInertiaForm } from '@inertiajs/vue3'
+import { InertiaForm, useForm as useInertiaForm } from '@inertiajs/vue3'
 import { watchEffect } from 'vue'
 
 export { client }
 
-export const useForm = <Data extends Record<string, unknown>>(method: RequestMethod|(() => RequestMethod), url: string|(() => string), inputs: Data, config: ValidationConfig = {}): any => {
+type Precognitive<Data extends object> = {
+    validating: boolean
+    touched: (name: keyof Data) => boolean
+    valid: (name: keyof Data) => boolean
+    invalid: (name: keyof Data) => boolean
```

Comment:
I went with this type because that's how the `laravel-precognition-vue` types were setup, but I don't actually think this is a good type. Error names might come back as single strings with dot-notation, which isn't the same as `keyof Data`. For example

```json
{
    "author": {
      "name": "Hello
   }
}
```
An error for the name will be indiicated with `"author.name"`.

For that reason, I'd probably actualyl suggest:
```suggestion
    touched: (name: string) => boolean
    valid: (name: string) => boolean
    invalid: (name: string) => boolean
```

Reply:
I'm gonna leave these for now and focus this PR just on adding the types to the Inertia flavour libraries.

Happy for these to up updated globally once we have typing everywhere.